proto/buf/alpha/registry/v1alpha1/repository.proto

```diff
@@ -202,3 +208,13 @@ message DeleteRepositoryByFullNameRequest {
 }
 
 message DeleteRepositoryByFullNameResponse {}
+
+message UpdateRepositoryDeprecationRequest{
+  string id = 1;
+  bool deprecated = 2;
+  string deprecation_message = 3;
+}
```

Comment:
The semantics here get a little blurred for the two different use cases: undeprecating vs. deprecating.

For deprecating, it makes sense - we require a non-empty `deprecation_message` and can return a `InvalidArgumentError` to the user if it's not set.

For undeprecating though, what does it mean to set `deprecated` to `false`, and give a non-empty `deprecation_message`? Will the `deprecation_message` just be dropped entirely and have no effect? Persisting it to the database wouldn't be useful since it would never be surfaced if `deprecated` is `false`.

With that said, it might be best to either:
* Model this as a `RepositoryState` (similar to the generic `UserState` proposed for [de]activating users).
  * Do we imagine any other states like `retracted` on the repository level that would actually cause an error to the user? If so, a more generic `RepositoryState` could be worthwhile.
* Split this out into two endpoints: `DeprecateRepository` and `UndeprecateRepository`, where each endpoint only exposes the minimal amount of fields necessary to perform the action.

The decision here might depend on how we want to model this in the `Module` definitions. If we ever want to "retract" specific module versions, it might make more sense to model the same ability for Repositories (i.e. the `RepositoryState` option). We could define a separate `ModuleState` enum so that the two states could evolve independently though.

What are your thoughts?

Reply:
I started out with DeprecateRepository and UndeprecateRepository as you suggested, but switched to UpdateRepositoryDeprecation for consistency after I saw the other UpdateRepository* rpcs. I'm happy to switch back to that.